api/src/api/notes.js

```diff
@@ -61,28 +62,47 @@ router.get(
       ],
     }).lean();
 
+    const memberIds = [
+      ...note.metaData.access.viewableBy,
+      ...note.metaData.access.editableBy,
+      ...note.metaData.referencedMembers,
+    ];
+    const uniqueMemberIds = [...new Set(memberIds)];
+    const mappedMembers = await memberFromId(uniqueMemberIds);
+
     // Replace all members ids with object that has id and name
-    note['metaData']['access']['viewableBy'] = await memberFromId(
-      note['metaData']['access']['viewableBy'],
+    note.metaData.access.viewableBy = note.metaData.access.viewableBy.map(
+      (member) => mappedMembers[member],
     );
-    note['metaData']['access']['editableBy'] = await memberFromId(
-      note['metaData']['access']['editableBy'],
+    note.metaData.access.editableBy = note.metaData.access.editableBy.map(
+      (member) => mappedMembers[member],
     );
-    note['metaData']['referencedMembers'] = await memberFromId(
-      note['metaData']['referencedMembers'],
+    note.metaData.referencedMembers = note.metaData.referencedMembers.map(
+      (member) => mappedMembers[member],
     );
 
     if (note.encrypt) {
-      const encryptionKey = req.user.oauthID + req.user.UIN;
-
-      try {
-        note.content = await decryptNote(note.content, encryptionKey);
-      } catch (err) {
-        return res.status(403).json({
-          success: false,
-          message: 'Unauthorized',
-        });
-      }
+      const partialEncryptionKey = req.user.oauthID + req.user.UIN;
+      const authedCredentials = req.session.authedCredentials;
```

Comment:
nit: `const { authedCredentials } = req.session` and re-use `authedCredentials` down below. You can do the same in `router.post('/', ...`